### ISSUE TITLE
Ansible requires ssh for localhost even with "--connection=local", "-c local" or "connection: local" in playbook file

### ISSUE THREAD

**Issue by grossmane** [ansible/ansible#12208]
##### Issue Type:
- Bug Report
##### Ansible Version:

ansible 1.9.2
  configured module search path = None
##### Ansible Configuration:

```
# inventory
127.0.0.1 
```
##### Environment:

CentOS 6.6
##### Summary:

this issue is similar to #11695

I have the case, that ansible want use ssh connection, if i run ansible-playbook with argument `"--connection=local"` or `"-c local"`.
If I set in playbook file `"connection: local"` we have this problem, too.

The only way, how it works is, if we set in inventory file `"ansible_connection=local"` or if we give an extra argument to ansible-playbook command: `"-e ansible_connection=local"`.
##### Steps To Reproduce:

```

---
- name: Test ping on localhost

  hosts: localhost
  connection: local

  tasks:

  - name: Ping!
    ping:
```

```
run: `ansible-playbook -v test.yml`
Remove the line connection: `local from test.yml`
run: `ansible-playbook -v test.yml --connection=local`
```
##### Expected Results:

Ansible being able to run locally without the need to have ssh installed.
##### Actual Results:

Ansible doesn't require ssh when ansible_connection=local is defined in the hosts file.
Ansible requires ssh for a playbook when run with flag --connection=local.
Ansible requires ssh for a playbook when declared connection: local in the playbook.

**Comment by jimi-c** [comment]
Hi, this is a duplicate of https://github.com/ansible/ansible/issues/11695, which has been closed and resolved (and will be included in 2.0 and 1.9.3).

Thanks!

**Comment by grossmane** [comment]
I don't understand, why this ticket is a duplicate of #11695.

The actual results in 11695 was:
Ansible requires ssh when ansible_connection=local is defined in the hosts file.
Ansible doesn't require ssh for a playbook when run with flag --connection=local.
Ansible doesn't require ssh for a playbook when declared connection: local in the playbook.

The actual results in 12208 was:
Ansible doesn't require ssh when ansible_connection=local is defined in the hosts file.
Ansible requires ssh for a playbook when run with flag --connection=local.
Ansible requires ssh for a playbook when declared connection: local in the playbook.

**Comment by jimi-c** [comment]
@grossmane because I believe the root causes were the same and addressed by the patch merged in to fix #11695. Is that not the case? Have you tested this against stable-1.9 and/or devel?
